### PR TITLE
V2.7.9 imu calibration fix

### PR DESCRIPTION
**Pull request by brentpicasso**
This fixes a number of things for IMU. This will feed into the 2.7.9 release.  Bench tested on MK1 and MK2.

1) calibration was broken when the channels were re-mapped (for alternate install orientation)

2) restored the natural orientations the IMU outputted (for some reason we tried to change it around a long time ago. I forgot why. Probably trying to be too clever. So, we'll let the IMU output what it thinks is positive and negative G's and Gyros, and say that's industry standard, dammit)

3) Everytime we wrote IMU configuration, it called imu_init, which created a new IMU task. Not good.  Created a imu_soft_init function to bypass this boot-up init.

### CODE REVIEW

File: src/imu/imu.c [hunk 1]
```diff
@@ -29,19 +29,6 @@ float imu_read_value(unsigned char imuChannel, ImuConfig *ac){
...
-		case IMU_CHANNEL_YAW:
-		case IMU_CHANNEL_PITCH:
-		case IMU_CHANNEL_ROLL:
-			scaledValue = -scaledValue;
-			break;
-		default:
-			break;
-	}
-
 	//now invert based on configuration
```
Comment (brentpicasso) [inline src/imu/imu.c]: I forget why I thought we needed to do this. I must've been drunk. killing it.

File: src/logger/loggerApi.c [hunk 2]
```diff
@@ -591,7 +591,7 @@ static void getImuConfigs(size_t channelId, void ** baseCfg, ChannelConfig ** ch
 }
 
 int api_setImuConfig(Serial *serial, const jsmntok_t *json){
-	int res = setMultiChannelConfigGeneric(serial, json, getImuConfigs, setImuExtendedField, imu_init);
+	int res = setMultiChannelConfigGeneric(serial, json, getImuConfigs, setImuExtendedField, imu_soft_init);
 	return res;
```
Comment (brentpicasso) [inline src/logger/loggerApi.c]: call soft init so we don't create a new IMU task every time we write new IMU config. >.>

File: src/imu/imu.c [hunk 3]
```diff
@@ -71,17 +58,24 @@ void imu_calibrate_zero(){
 		int zeroValue = g_imu_filter[physicalChannel].current_value;
 		//adjust for gravity
 		float countsPerUnit = imu_device_counts_per_unit(physicalChannel);
-		if (c->physicalChannel == IMU_CHANNEL_Z) zeroValue-= (countsPerUnit * (c->mode != MODE_IMU_INVERTED ? 1 : -1));
+		if (i == IMU_CHANNEL_Z) zeroValue -= (countsPerUnit * (c->mode != MODE_IMU_INVERTED ? 1 : -1));
 		c->zeroValue = zeroValue;
```
Comment (brentpicasso) [inline src/imu/imu.c]: adjust for gravity on the actual mapped channel.

File: src/imu/imu.c [hunk 4]
```diff
@@ -71,17 +58,24 @@ void imu_calibrate_zero(){
 		int zeroValue = g_imu_filter[physicalChannel].current_value;
 		//adjust for gravity
 		float countsPerUnit = imu_device_counts_per_unit(physicalChannel);
-		if (c->physicalChannel == IMU_CHANNEL_Z) zeroValue-= (countsPerUnit * (c->mode != MODE_IMU_INVERTED ? 1 : -1));
+		if (i == IMU_CHANNEL_Z) zeroValue -= (countsPerUnit * (c->mode != MODE_IMU_INVERTED ? 1 : -1));
```
Comment (Jeff-Ciesielski) [inline src/imu/imu.c]: This statement is entirely too tricky.

File: src/imu/imu.c [hunk 5]
```diff
@@ -71,17 +58,24 @@ void imu_calibrate_zero(){
...
 		c->zeroValue = zeroValue;
 	}
 }
 
 int imu_init(LoggerConfig *loggerConfig){
 	imu_device_init();
-	init_filters(loggerConfig);
+    init_filters(loggerConfig);
 	return 1;
```
Comment (Jeff-Ciesielski) [inline src/imu/imu.c]: Broken indentation

File: src/imu/imu.c [hunk 6]
```diff
@@ -71,17 +58,24 @@ void imu_calibrate_zero(){
...
 }
 
+int imu_soft_init(LoggerConfig *loggerConfig){
+    init_filters(loggerConfig);
+    return 1;
+}
```
Comment (Jeff-Ciesielski) [inline src/imu/imu.c]: Any reason we are returning a value here?

File: src/imu/imu.c (hunk 4 above)
Comment (brentpicasso) [inline src/imu/imu.c]: I think we can put the true part of the if statement it it's own block, per our convention.

But, are ternary operators in general "too tricky"?

File: src/imu/imu.c (hunk 5 above)
Comment (brentpicasso) [inline src/imu/imu.c]: want to run everything through a code formatter after this is done.

File: src/imu/imu.c (hunk 6 above)
Comment (brentpicasso) [inline src/imu/imu.c]: It's a standard convention for HAL / device init methods, even if it always succeeds.

File: src/imu/imu.c (hunk 4 above)
Comment (Jeff-Ciesielski) [inline src/imu/imu.c]: Case dependant methinks.  I really dislike them as a matter of preference (Kernel standard states "Don't put multiple statements on a single line unless you have something to hide"), but as with most things in C, they have their place if used sparingly.

File: src/imu/imu.c (hunk 6 above)
Comment (Jeff-Ciesielski) [inline src/imu/imu.c]: I see now why you are doing this.  Looks icky out of context, but this function is effectively adhering to an interface.

File: src/imu/imu.c [hunk 7]
```diff
@@ -64,24 +51,35 @@ static void imu_flush_filter(size_t physicalChannel){
...
 		size_t physicalChannel = c->physicalChannel;
 		imu_flush_filter(physicalChannel);
 		int zeroValue = g_imu_filter[physicalChannel].current_value;
-		//adjust for gravity
 		float countsPerUnit = imu_device_counts_per_unit(physicalChannel);
-		if (c->physicalChannel == IMU_CHANNEL_Z) zeroValue-= (countsPerUnit * (c->mode != MODE_IMU_INVERTED ? 1 : -1));
+		if (logicalChannel == IMU_CHANNEL_Z){ //adjust for gravity
+		    if (c->mode == MODE_IMU_INVERTED){
+		        countsPerUnit = -countsPerUnit;
+		    }
```
Comment (brentpicasso) [inline src/imu/imu.c]: Ok, made less tricky.

File: src/imu/imu.c (hunk 4 above)
Comment (brentpicasso) [inline src/imu/imu.c]: pushed an update to make less tricky.

**Pull request merged**